src/main/java/com/avispl/symphony/dal/communicator/aggregator/ZoomRoomsAggregatorCommunicator.java

```diff
@@ -0,0 +1,433 @@
+        if (roomMetrics != null && roomMetrics.isArray()) {
+            for (JsonNode roomMetric: roomMetrics) {
+                HashMap<String, String> roomProperties = new HashMap<>();
+                aggregatedDeviceProcessor.applyProperties(roomProperties, roomMetric, "ZoomRoomMetrics");
+                roomMetricsProperties.put(roomProperties.get("id"), roomProperties);
+            }
+        }
+
+        Long currentTime = new Date().getTime();
+        for (AggregatedDevice room : rooms) {
+            String roomId = room.getDeviceId();
+            room.getProperties().putAll(roomMetricsProperties.get(roomId));
+
+            Map<String, String> settingsProperties = new HashMap<>();
+            List<AdvancedControllableProperty> settingsControls = new ArrayList<>();
+
+            JsonNode meetingSettings = retrieveRoomMeetingSettings(roomId);
+            aggregatedDeviceProcessor.applyProperties(settingsProperties, settingsControls, meetingSettings, "RoomMeetingSettings");
+//            JsonNode alertSettings = retrieveRoomAlertSettings(roomId);
+//            aggregatedDeviceProcessor.applyProperties(settingsProperties, settingsControls, meetingSettings, "RoomAlertSettings");
+
+            room.getProperties().putAll(settingsProperties);
+            room.getControllableProperties().addAll(settingsControls);
+            JsonNode devices = retrieveRoomDevices(roomId);
+            if (devices != null && devices.isArray()) {
+                for (JsonNode deviceNode : devices) {
+                    Map<String, String> properties = new HashMap<>();
+                    aggregatedDeviceProcessor.applyProperties(properties, deviceNode, "RoomDevice");
+                    // TODO add properties to aggregated device
+
+                    for (Map.Entry<String, String> entry : properties.entrySet()) {
+                        room.getProperties().put(String.format("RoomDevice_%s_%s", properties.get("Info#ID"), entry.getKey()), entry.getValue());
+                    }
+                }
+            }
+
+            if (room.getProperties().get("Metrics#RoomStatus").equals("InMeeting")) {
+                room.getProperties().put("RoomControls#EndCurrentMeeting", "");
+                room.getControllableProperties().add(createButton("RoomControls#EndCurrentMeeting", "End", "Ending...", 0L));
+
+                room.getProperties().put("RoomControls#LeaveCurrentMeeting", "");
+                room.getControllableProperties().add(createButton("RoomControls#LeaveCurrentMeeting", "Leave", "Leaving...", 0L));
+            }
+
+            if (!room.getProperties().get("Metrics#RoomStatus").equals("Offline")) {
+                room.getProperties().put("RoomControls#RestartZoomRoomsClient", "");
+                room.getControllableProperties().add(createButton("RoomControls#RestartZoomRoomsClient", "Restart", "Restarting...", 0L));
+            }
+
+            room.setTimestamp(currentTime);
+        }
+    }
+
+    /**
+     *

```

Comment:
Guessing you are still working on javadocs?

FYI, we don't necessarily need javadocs for private methods. But, if there is logic in there that needs to be reverse engineered, or custom code parsing some URL, they are very useful :)

Reply:
Yes, javadocs will be added later, but I can address it as a part of this pr